### PR TITLE
Several fixes for using icalendar without ActiveSupport

### LINKED ISSUE #288: icalendar can't handle timezones coming from Apple's iCloud .ical files
.ical calendars can be downloaded from Apple's iCloud, and icalendar somehow can't handle the timezone infomration contained within them.    For example, despite the ical data for `DTSTART` being:

    DTSTART;TZID=America/Los_Angeles:20231025T163000

icalendar treats this as being in the UTC timezone.

Sample (redacted) .ics file: [raw.txt](https://github.com/icalendar/icalendar/files/13434597/raw.txt)

Test ruby script to show the bug:
```
require 'icalendar'

cals = Icalendar::Calendar.parse(File.open("raw.txt"))

first = cals.first
first.events.each { |e|
  #pp e
  print "dtstart = ", e.dtstart, "\n"
}
```
Running the script produces the incorrect timezone:
```
dtstart = 2023-10-25T16:30:00+00:00
```

### PR DESCRIPTION
**Pull request by rahearn**
Changes:

* Defers calculating the offset until later in the process when VTIMEZONE hasn't been defined yet while parsing events.
* Fixes calculation of VTIMEZONE occurrence when the event is before the most recent timezone rules.

closes #288

### CODE REVIEW

File: spec/fixtures/reversed_timezone.ics [hunk 1]
```diff
@@ -0,0 +1,143 @@
+BEGIN:VCALENDAR
+VERSION:2.0
+X-WR-CALNAME:My Calendar
+X-APPLE-CALENDAR-COLOR:#44A703
+BEGIN:VEVENT
+CREATED:20231026T211915Z
+DTEND;TZID=NONSTANDARDTZ:20231025T180000
+DTSTAMP:20231026T211916Z
+DTSTART;TZID=NONSTANDARDTZ:20231025T163000
```
Comment (rahearn) [inline spec/fixtures/reversed_timezone.ics:9]: using this `TZID` proves that this solution should work whether or not `tzinfo` or `activesupport` are being used.

**Comment by darrylo** [comment]
Unfortunately, this has another bug:
```
 comparison of IceCube::Occurrence with nil failed (ArgumentError)
      possible.sort_by(&:first).last
```
Run the previous `bug.rb` with this [`raw.txt`](https://github.com/icalendar/icalendar/files/13467793/raw.txt) (the bug is triggered by the second VEVENT).

Also, can we get an option to suppress the many, **many** messages like:
```
W, [2023-11-26T11:04:24.273201 #2094]  WARN -- : Ignoring timezone UTC for time 2023-11-26T19:04:24+00:00
W, [2023-11-26T11:06:57.786737 #2110]  WARN -- : Ignoring timezone America/Los_Angeles for time 2023-10-25T18:00:00+00:00
```
Thanks.

**Comment by rahearn** [comment]
@darrylo good catch - I've fixed the occurrence bug and downgraded the warning log messages to `info` (which aren't output by default, but are easy enough to turn on without all of the `debug` noise)

**Comment by darrylo** [comment]
OK, seems good now.  Thanks.

**Comment by darrylo** [comment]
While this "works", it can really screw with the `Icalendar::Values::DateTime` datatype, which becomes a different class depending on what's installed.  For example, I can use `.localtime` on the `.dtstamp` value if I have activesupport installed.  Without activesupport installed, that method does not exist, and it's a real PITA to convert the time to localtime.

Installing activesupport is just going to be the universal workaround.

**Comment by rahearn** [comment]
> While this "works", it can really screw with the `Icalendar::Values::DateTime` datatype, which becomes a different class depending on what's installed. For example, I can use `.localtime` on the `.dtstamp` value if I have activesupport installed. Without activesupport installed, that method does not exist, and it's a real PITA to convert the time to localtime.
> 
> Installing activesupport is just going to be the universal workaround.

This was already an issue even before this change, though I agree one that I need to take a closer look at.

**Pull request merged**